### ISSUE TITLE
Does not appear to work in Vite

### ISSUE THREAD

**Issue by aral** [JonasKruckenberg/rollup-plugin-sri#99]
Hi there, 

First off, thank you for making and sharing this :)

I’ve been trying to get the plugin to work with [Vite](https://vitejs.dev/) but it doesn’t seem to have an affect.

Thoughts? Should I open this issue in Vite instead?

## To reproduce

1. npm init @vitejs/app
2. Name it _vite-sri-test_ and choose `vanilla` for template to create the simplest possible app
3. `cd vite-sri-test`
4. `npm i`
5. `npm run build` and note the `index.html` file output in `dist/`:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" type="image/svg+xml" href="/assets/favicon.17e50649.svg" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
  <script type="module" crossorigin src="/assets/index.b2293eba.js"></script>
  <link rel="stylesheet" href="/assets/index.ccce2ca3.css">
</head>
  <body>
    <div id="app"></div>
    
  </body>
</html>
```

6. `npm i rollup-plugin-sri`
7. Create `vite.config.js` with the following content:

```js
import sri from 'rollup-plugin-sri'
import { defineConfig } from 'vite'

export default defineConfig({
  plugins: [sri()]
})
```
8. `npm run build` and note the `index.html` file output in `dist/`:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" type="image/svg+xml" href="/assets/favicon.17e50649.svg" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
  <script type="module" crossorigin src="/assets/index.b2293eba.js"></script>
  <link rel="stylesheet" href="/assets/index.ccce2ca3.css">
</head>
  <body>
    <div id="app"></div>
    
  </body>
</html>
```
9. Add the `@rollup/plugin-html` and try with that (`npm i @rollup/plugin-html` + add the import line (`import html from '@rollup/plugin-html'` and change `plugins: [sri()]` to `plugins: [html(), sri()`) and try `npm run build` again. Note the output:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" type="image/svg+xml" href="/assets/favicon.17e50649.svg" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
  <script type="module" crossorigin src="/assets/index.b2293eba.js"></script>
  <link rel="stylesheet" href="/assets/index.ccce2ca3.css">
</head>
  <body>
    <div id="app"></div>
    
  </body>
</html>
```

10. Add the plugins to `build.rollupOptions.plugins` instead and note the resulting HTML (this time it’s different but still doesn’t include the integrity hashes):

```html
<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>Rollup Bundle</title>
    <link href="assets/index.ccce2ca3.css" rel="stylesheet">
  </head>
  <body>
    <script src="assets/index.b2293eba.js" type="module"></script>
  </body>
</html>
```

## What should happen

Integrity hashes should be included

## What actually happens

Integrity hashes are not included

## More details

This output from Vite could be related:

```
rendering chunks (1)...The emitted file "index.html" overwrites a previously emitted file of the same name.
```

Version of Vite: 2.1.4
Version of rollup-plugin-sri: 1.2.7
Version of @rollup/plugin-html: 0.2.3

**Comment by JonasKruckenberg** [comment]
Thanks for the heads up! The TLDR version is: This is a Vite particularity, but a feature not a bug.
The long version is as follows: Back when I wrote this plugin, the only way to get the full content of each asset (even those generated by other plugins) was to read them back from disk in the _writeBundle_ hook. This is not elegant but it works in rollup. In Vite however, the writeBundle hook never get's called (A Vite particularity) another problem is that the index.html gets treated differently than other assets and there is a separate hook for that file. 

So to recap, I didn't expect the plugin to work out of the box, but since Vite seems to be a popular choice and because I'm a lot more familiar with Vite now, I might revisit this plugin and see if I can add Vite support 👍

**Comment by JonasKruckenberg** [comment]
I've seen you forked the repo, If you already found a solution to make this work with Vite, feel free to submit a PR!

**Comment by JonasKruckenberg** [comment]
Jep that seems to be a better option, after looking into the issue a bit more it seems the problem isthat vite puts assets into the assets folder without passing that info via “native” rollup hooks. 

The solution would be to access the base attribute of the vite config but that'd add vite-specific code to a rollup plugin, so yeah.

I thinkfor documentation purposes this issue should be closed once your plugin goes live?

**Comment by JonasKruckenberg** [comment]
Update! 
> the problem is that vite puts assets into the assets folder without passing that info via “native” rollup hooks.

Nevermind actually! [Line 64](https://github.com/JonasKruckenberg/rollup-plugin-sri/blob/0968c8afb81a0b5643941074d2b3310e8243e6de/index.ts#L64) seems to be the culprit, and just removing that seems to make the plugin work with Vite no problem! 
I honestly don't remember what that was for and I will definitely do some more investigation and compatibility tests, but it seems that this fixes it.

However, I'd be interested in your improvements to the idea maybe we can even merge those back into this project so we're bundling our forces.

Anyway, awesome issue by the way! I've not seen such a thorough bug report in a long time 👍

**Comment by JonasKruckenberg** [comment]
Yeah don't know, somatic release is always a little too excited ;) But I'm fairly confident the released version (1.2.8) should work with Vite but if you find the time I'd be awesome if you could confirm that it indeed works!

**Comment by aral** [comment]
@JonasKruckenberg Happy to hear it. Just tested and it works well for local paths. For remote URLs (in both scripts and stylesheets), I get the following errors:

```
ENOENT: no such file or directory, open '/home/aral/sandbox/vite-sri-test-with-rollup-plugin/dist/https:/ar.al/index.js'
ENOENT: no such file or directory, open '/home/aral/sandbox/vite-sri-test-with-rollup-plugin/dist/https:/ar.al/style.css'
```

Looks like it’s treating them like local paths.

(In the meanwhile, I did write a plugin specifically for Vite that does the same thing for index.html which I’ll probably be using myself as it’s in JS and easier to maintain for me. I’ll also be linking to yours in the comments/readme as its a much more generic solution.) :)

**Comment by JonasKruckenberg** [comment]
Thanks for the feedback! I didn't catch that during testing so it's great to hear.

I'm looking forwards to seeing your plugin as well and might link to yours too!

**Comment by aral** [comment]
Hey Jonas, it’s out: https://github.com/small-tech/vite-plugin-sri

Much less configurable and limited than yours and unashamedly inspired by it :) Thanks again for making this and sharing it + if you need me to test anything else, just shout.
